migrate/migrations/000007_add-posts-table.up.sql

```diff
@@ -0,0 +1,31 @@
+CREATE TABLE posts (
+    id INT UNSIGNED AUTO_INCREMENT PRIMARY KEY,
+    created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+    updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
+
+    type TINYINT UNSIGNED NOT NULL CHECK (type IN (0,1,2,3,4)),
```

Comment:
I'm using `TINYINT` instead of `BIT` - they would both take up a byte of space, and I've read that `BIT` is difficult to work with

Reply:
I'm not sure about using the `CHECK` here? Maybe we leave this at the application level instead? 5 types was to have enough for growth

Reply:
Yeh I don't think `BIT` would be suitable for this if we plan on having more than 2 types.

I'm not sure on the `CHECK`, were probably going to end up doing this at the application level anyways right? But i suppose it does guarantee integrity. I guess it's just making sure its clear that allowing a new post type means updating the application level checks as well as the schema (which should be just dropping the constraint and re-applying?)